server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByWithValueFilterDataSet.java

```diff
@@ -37,30 +37,21 @@
 import org.apache.iotdb.tsfile.read.common.Path;
 import org.apache.iotdb.tsfile.read.common.RowRecord;
 import org.apache.iotdb.tsfile.read.expression.IExpression;
+import org.apache.iotdb.tsfile.read.common.TimeColumn;
 import org.apache.iotdb.tsfile.read.query.timegenerator.TimeGenerator;
 
 public class GroupByWithValueFilterDataSet extends GroupByEngineDataSet {
 
   private List<IReaderByTimestamp> allDataReaderList;
   private GroupByPlan groupByPlan;
   private TimeGenerator timestampGenerator;
-  /**
-   * cached timestamp for next group by partition.
-   */
-  private long timestamp;
-  /**
-   * if this object has cached timestamp for next group by partition.
-   */
-  private boolean hasCachedTimestamp;
 
+  private TimeColumn timeColumn = new TimeColumn();
   /**
    * group by batch calculation size.
    */
   protected int timeStampFetchSize;
```

Comment:
can be private. And also other methods in this class

Reply:
done